pkg/scheduler/framework/plugins/volumezone/volume_zone.go

```diff
@@ -261,22 +268,206 @@ func getErrorAsStatus(err error) *framework.Status {
+	return framework.QueueSkip
+}
+
+// isSchedulablerNode checks whether the node can schedule the pod.
+func (pl *VolumeZone) isSchedulableNode(podPVTopologies []pvTopology, node *v1.Node) bool {
+	hasAnyNodeConstraint := false
+	for _, topologyLabel := range topologyLabels {
+		if _, ok := node.Labels[topologyLabel]; ok {
+			hasAnyNodeConstraint = true
+			break
+		}
+	}
+
+	if !hasAnyNodeConstraint {
+		// The node has no zone constraints, so we're OK to schedule.
+		// This is to handle a single-zone cluster scenario where the node may not have any topology labels.
+		return true
+	}
+
+	for _, pvTopology := range podPVTopologies {
+		v, ok := node.Labels[pvTopology.key]
+		if !ok || !pvTopology.values.Has(v) {
+			return false
+		}
+	}
+	return true
+}
+
+// isSchedulableAfterPersistentVolumeClaimAdded is invoked whenever a PersistentVolumeClaim added. It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *VolumeZone) isSchedulableAfterPersistentVolumeClaimAdded(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	_, addedPVC, err := util.As[*v1.PersistentVolumeClaim](nil, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterPersistentVolumeClaimAdded", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
+	}
+
+	for i := range pod.Spec.Volumes {
+		volume := pod.Spec.Volumes[i]
+		if volume.PersistentVolumeClaim == nil {
+			continue
+		}
+		pvcName := volume.PersistentVolumeClaim.ClaimName
+		if (addedPVC.Name == pvcName) && (pod.Namespace == addedPVC.Namespace) {
+			logger.V(4).Info("PVC was created, and matches the pod's PVC", "pod", klog.KObj(pod), "PVC", klog.KObj(addedPVC))
+			return framework.QueueAfterBackoff
+		}
+	}
+
+	logger.V(4).Info("PVC was created, but it doesn't match the pod's PVC", "pod", klog.KObj(pod), "PVC", klog.KObj(addedPVC))
+	return framework.QueueSkip
+}
+
+// isSchedulableAfterPersistentVolumeChange is invoked whenever a PersistentVolume added or updated. It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *VolumeZone) isSchedulableAfterPersistentVolumeChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	originalPV, modifiedPV, err := util.As[*v1.PersistentVolume](nil, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterPersistentVolumeChange", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
+	}
+	originalPVTopologies := make([]pvTopology, 0)
+	if originalPV != nil {
+		originalPVTopologies, _ = pl.getUpdatedPVsTopologies(logger, pod, originalPV)
 	}
+	modifiedPVTopologies, isSchedulable := pl.getUpdatedPVsTopologies(logger, pod, modifiedPV)
+	if isSchedulable && reflect.DeepEqual(originalPVTopologies, modifiedPVTopologies) {
```

Comment:
I think we can't simply use `reflect.DeepEqual` to compare two slice(it will compare the order as well), and in this case, we don't care the order in []pvTopology. Maybe use something like `[cmpopts.SortSlices]` or what.

Reply:
sorting is good to me